src/astgen.zig

```diff
@@ -470,30 +466,27 @@ pub fn expr(mod: *Module, scope: *Scope, rl: ResultLoc, node: ast.Node.Index) In
             return rvalue(mod, scope, rl, result, node);
         },
         .optional_type => {
-            if (true) @panic("TODO update for zir-memory-layout");
-            const src = token_starts[main_tokens[node]];
+            const src_token = tree.firstToken(node);
+
             const operand = try typeExpr(mod, scope, node_datas[node].lhs);
-            const result = try addZIRUnOp(mod, scope, src, .optional_type, operand);
-            return rvalue(mod, scope, rl, result);
+            const result = try gz.addUnTok(.optional_type, operand, src_token);
+            return rvalue(mod, scope, rl, result, node);
         },
         .unwrap_optional => {
-            if (true) @panic("TODO update for zir-memory-layout");
+            const src_token = tree.firstToken(node);
+
             const src = token_starts[main_tokens[node]];
             switch (rl) {
-                .ref => return addZIRUnOp(
-                    mod,
-                    scope,
-                    src,
+                .ref => return gz.addUnTok(
                     .optional_payload_safe_ptr,
                     try expr(mod, scope, .ref, node_datas[node].lhs),
+                    src_token,
                 ),
-                else => return rvalue(mod, scope, rl, try addZIRUnOp(
-                    mod,
-                    scope,
-                    src,
+                else => return rvalue(mod, scope, rl, try gz.addUnTok(
                     .optional_payload_safe,
                     try expr(mod, scope, .none, node_datas[node].lhs),
-                )),
+                    src,
```

Comment:
looks like passing a byte offset where a token index is expected. Be really careful about these - the compiler can't catch these errors for us unless we figure out how to make the types be distinct!

I think probably all the `token_starts` from this entire file should be deleted.